napari/layers/labels/labels.py

```diff
@@ -1445,9 +1421,9 @@ def data_setitem(self, indices, value, refresh=True):
 
         Parameters
         ----------
-        indices : tuple of int, slice, or sequence of int
-            Indices in data to overwrite. Can be any valid NumPy indexing
-            expression [1]_.
+        indices : tuple of arrays of int
+            Indices in data to overwrite. Must be a tuple of arrays of length
+            equal to the number of data dimensions. (Fancy indexing in [1]_).
```

Comment:
As it is mentioned here as important, then maybe we could add a type annotation to the argument, to allow pycharm, vs code plugins etc, to highlight potential problems.

Reply:
@Czaki since numpy arrays aren't yet generic types (I think?), ie I can't write `Tuple[np.ndarray[int], ...]`, and also since eventually we want to expand to allow arbitrary indexing expressions, I think it's best to leave this flexible for now.